.github/workflows/ci.yaml

```diff
@@ -11,7 +11,27 @@ on:
       - "feature/**"
 
 jobs:
-  build:
+  pre_build:
+    runs-on: self-hosted
+    #runs-on: ubuntu-latest
+
+    outputs:
+      output_tag: ${{ steps.tag_wp.outputs.tag }}
+      output_owner_lc: ${{ steps.owner_lowercase.outputs.lowercase }}
+
+    steps:
+      - id: tag_wp
+        run: echo "tag=$(date +"%Y.%m.%d.%H.%M.%S")-${{ github.run_id }}" >> $GITHUB_OUTPUT
+
+      #- id: owner_lowercase
+        #run: echo "owner_lc=${{ github.repository_owner }} | tr '[:upper:]' '[:lower:]'" >> $GITHUB_OUTPUT
+
+      - id: owner_lowercase
+        uses: ASzc/change-string-case-action@v5
```

Comment:
owner should be same organization

Reply:
and if you need to make it lowercase, do not use custom step, bash is fine enough